augmentation/augmentation.hpp

```diff
@@ -170,7 +213,119 @@ class Augmentation
     }
   }
 
-  //! Locally held augmentations and transforms that need to be applied.
+/**
+   * Function to determine if augmentation has horizontal-flip function.
+   *
+   * @param augmentation Optional argument to check if a string has
+   *                     horizontal-flip substring.
+   */
+  bool HasHorizontalFlipParam(const std::string& augmentation = "")
+  {
+    if (augmentation.length())
+      return augmentation.find("horizontal-flip") != std::string::npos;
+
+
+    // Search in augmentation vector.
+    for(size_t i=0; i<augmentations.size(); i++)
+    {
+      if(augmentations[i].find("horizontal-flip") != std::string::npos)
+          return true;
+    }
+    return false;
+
+  }
```

Comment:
I don't think we need this if we use a map instead.

Reply:
yes